File: alarmer/c_alarmer/console.cpp
#include "console.h"


#include <iostream>
#include <fstream>
#include <cstdarg>
#include <filesystem>
#include <vector>


std::string get_document_dir_path()
{
  std::filesystem::path sub_current_dir_path = std::filesystem::current_path();
  std::string current_dir_path = sub_current_dir_path.string();

  char slash = '/';
  size_t pos = current_dir_path.rfind(slash);
  if (pos != std::string::npos) {
    current_dir_path.erase(pos);
  }

  std::string document_dir_path = current_dir_path + "/c_documents/";

  return document_dir_path;
}


std::string find_document(const std::string& document_file_name)
{
  std::string document_dir_path = get_document_dir_path();
  std::string document_file_path = document_dir_path + document_file_name;

  return document_file_path;
}


std::string get_document(const std::string& document_file_name, const std::string& color)
{
  std::string document = find_document(document_file_name);

  std::ifstream document_file(document);
  if (!document_file) {
    std::cout << "cannot find " << document << std::endl;
    exit(1);
  };
  std::string content((std::istreambuf_iterator<char>(document_file)), std::istreambuf_iterator<char>());
  std::string message(content);
  std::string splitter(60, '=');
  message = color + splitter + "\n" + message + "\n" + splitter + "\033[0m";

  return message;
}


std::string make_message_from_document(
  const std::string& document_file_name,
  const std::string& color,
  const std::vector<std::string>& replace_word
)
{
  std::string message = get_document(document_file_name, color);
  for (const auto& word : replace_word) {
    message.replace(message.find("$replace"), 8, word);
  };

  return message;
}

File: alarmer/views/console.py
import os
import string
import termcolor


def get_document_dir_path():
  """documentsディレクトリのパスを返す
  
  Returns:
    document_dir_path (str): documentsディレクトリのパス
  """
  base_dir_path = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
  document_dir_path = os.path.join(base_dir_path, 'documents')

  return document_dir_path


class NoDocumentError(Exception):
  """ドキュメントが見つからない場合のエラー"""
  pass


def find_document(document_file_name):
  """documentディレクトリのなかから必要なドキュメントを見つける
  
  Args:
    document_file_name (str): 必要なドキュメントのファイルネーム
    
  Returns:
    document_file_path (str): ドキュメントのファイルパス
    
  Raises:
    NoDocumentError: 必要なドキュメントが見つからない場合に発生する
  """
  document_dir_path = get_document_dir_path()
  document_file_path = os.path.join(document_dir_path, document_file_name)

  if not os.path.exists(document_file_path):
    raise NoDocumentError(f'cannot find {document_file_path}')

  return document_file_path


def get_document(document_file_name, color):
  """documentsディレクトリのなかのテキストファイルの内容を
     任意に変更できるかたちで返す
     
  Args:
    document_file_name (str): ドキュメントのファイルネーム
    color: (str): ターミナルに出力するときの文字の色
        詳細: https://pypi.python.org/pypi/termcolor
        
  Returns:
    string.Template: 任意に変更できるようになった、
    documentsディレクトリのなかのテキストファイルの内容
  """
  document = find_document(document_file_name)

  with open(document, 'r') as document_file:
    content = document_file.read()
    splitter = '=' * 60
    contents = f'{splitter}\n{content}\n{splitter}\n'
    contents = termcolor.colored(contents, color)

  return string.Template(contents)


def make_message_from_document(document_file_name, color, 
sub_error_message = '', sub_what_do = '', sub_min_time = '', sub_max_time = '', 
sub_now = '', sub_ok_or_ng_message = '', sub_start_alarm_time = '',
sub_stop_alarm_time = '',  sub_interval = ''):
  """documentsディレクトリのなかのテキストファイル
     の内容を書き換え、臨機応変にメッセージを作る
     
  Args:
    document_file_name (str): 開きたいファイルの名前
    color: (str): ターミナルに出力するときの文字の色
        詳細: https://pypi.python.org/pypi/termcolor
    以下 (str): ファイルの内容を書き換えるための文
    
  Returns:
    message (str): 作成したメッセージ
  """

  message = get_document(document_file_name, color)

  message = message.substitute(
    error_message = sub_error_message,
    what_do = sub_what_do,
    min_time = sub_min_time,
    max_time = sub_max_time,
    now = sub_now,
    ok_or_ng_message = sub_ok_or_ng_message,
    start_alarm_time = sub_start_alarm_time,
    stop_alarm_time = sub_stop_alarm_time,
    interval = sub_interval,
  )

  return message